src/ChainOfResponssibility/Poker/Validators/HandValidators/StraightFlushValidation.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Linq;
+
+namespace ChainOfResponssibility.Poker.Validators.HandValidators
+{
+    public class StraightFlushValidation : HandValidation
+    {
+        protected override string IsValid(Card[] hand)
+        {
+            var gr = hand.GroupBy(x => x.Suite);
+            var flush = gr.FirstOrDefault(x => x.Count() == 5);
+            if (flush != null)
+            {
+                var Value2Index = hand.Select((i, j) => i.GetIntValue() - j).Distinct();
+                if (!Value2Index.Skip(1).Any())// consecutive
+                    return "You have a Straight Flush with " + string.Join(", ", hand.Select(x => x.ToString()));
+
+                //special case for A,2,3,4,5
```

Comment:
maybe for these cases we could use tree of responsibility?  given the fact that for a certain Hand you can have multiple cases. so instead of having one Successor, we have a collection of Successors. Each Successor will check for a combination of the cards from a specific hand. In this way, if a new combination on a certain hand will be created, we only need to create a new HandCombinationValidator for that scenario, add it to the Successors collection and that's it